### PR TITLE
upgrade max level fix, user resources fix

### PR DESCRIPTION
**Pull request by barakavraham**

### CODE REVIEW

File: app/models/army.py [hunk 1]
```diff
@@ -64,3 +64,10 @@ def get_current_level_num(self, upgrade_name):
     def add_level(self, upgrade_name):
         current_level = self.get_current_level_num(upgrade_name)
         setattr(self, upgrade_name, current_level + 1)
+
+    def is_max_level(self, upgrade_name):
+        current_level = getattr(self, upgrade_name)
+        return current_level == TECH_UPGRADES[upgrade_name].max_level
+
+
```
Comment (daniel-atbay) [inline app/models/army.py]: too many blank lines bitch

File: app/static/js/shop.js [hunk 2]
```diff
@@ -46,6 +46,8 @@
 
         function setButtonPrices($btn, prices) {
             let $prices = $btn.parents('.card').find('.prices');
+            for (let resource in {metal: 0, wood: 0, coin: 0})
```
Comment (daniel-atbay) [inline app/static/js/shop.js]: hell no

File: app/templates/shop/index.html [hunk 3]
```diff
@@ -71,64 +71,65 @@
                         </div>
                     </div>
                 </div>
-            {% endfor %}
+                {% endfor %}
```
Comment (daniel-atbay) [inline app/templates/shop/index.html]: i think this needs to go back, no?

File: app/templates/shop/index.html [hunk 4]
```diff
@@ -71,64 +71,65 @@
                         </div>
                     </div>
                 </div>
-            {% endfor %}
+                {% endfor %}
         </div>
     </div>
     <div class="container upgrades-container" style="display: none;">
-        <div class="row">
+            <div class="row">
```
Comment (daniel-atbay) [inline app/templates/shop/index.html]: this also needs to go back, no?

File: app/utils/shop.py [hunk 5]
```diff
@@ -13,13 +14,24 @@ def price(self, amount=1):
         return {item: self.prices[item] * amount for item in self.prices}
 
 
+
 class UpgradableItem:
-    def __init__(self, level_1: ShopItem, level_2: ShopItem, level_3: ShopItem, level_4: ShopItem, level_5: ShopItem):
+    def __init__(self, level_1: ShopItem=None, level_2: ShopItem=None,
```
Comment (daniel-atbay) [inline app/utils/shop.py]: do:
```suggestion
    def __init__(self, level_1: ShopItem=None, level_2: ShopItem=None,
    def __init__(self,
                 ShopItem = None, level_2: ShopItem = None,
```
And remember spaces on the sides of the `=`

**Review by daniel-atbay — APPROVED** [review]

File: app/models/army.py (hunk 1 above)
Comment (barakavraham) [reply app/models/army.py]: right bitch

File: app/static/js/shop.js (hunk 2 above)
Comment (barakavraham) [reply app/static/js/shop.js]: shhhh

File: app/templates/shop/index.html (hunk 3 above)
Comment (barakavraham) [reply app/templates/shop/index.html]: positive

File: app/templates/shop/index.html (hunk 4 above)
Comment (barakavraham) [reply app/templates/shop/index.html]: positive

File: app/utils/shop.py (hunk 5 above)
Comment (barakavraham) [reply app/utils/shop.py]: alright

Comment (daniel-atbay) [reply app/utils/shop.py]: danke bitch

**Review by daniel-atbay — APPROVED** [review]

**Pull request merged**